src/Features/LanguageServer/Protocol/Features/Diagnostics/DocumentAnalysisExecutor.cs

```diff
@@ -96,7 +96,7 @@ public async Task<IEnumerable<DiagnosticData>> ComputeDiagnosticsAsync(Diagnosti
                 // We will count generator diagnostics as semantic diagnostics; some filtering to either syntax/semantic is necessary or else we'll report diagnostics twice.
                 if (kind == AnalysisKind.Semantic)
                 {
-                    var generatorDiagnostics = await textDocument.Project.GetSourceGeneratorDiagnosticsAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
The other approach here would be to have _this_ method do the remoting to the other process, so that way any other callers get the same benefit. I'd imagine it would work basically the same way, just the methods are on a different interface but that's about it. I'm fine with this as it is though since it's a bit more clear where the generators are running, but it does leave a gotcha for future callers.

Reply:
I did try that route first, and hit a roadblock - maybe because I was trying to reuse the existing method on IRemoteDiagnosticAnalyzerService for source generator diagnostics, I can't recall now. Definitely feel free to make this follow-up cleanup change if we feel that hardens the code base.